listener/mqueue.py

```diff
@@ -0,0 +1,38 @@
+import os
+import json
+import logging
+import datetime
+from kafka import KafkaConsumer, KafkaProducer
+
+BOOTSTRAP_SERVERS = os.getenv('MESSAGE_SERVICE', 'kafka:9092')
+GROUP_ID = os.getenv('MESSAGE_GROUP', 'vulnerability')
+ENGINE_RESULTS_TOPIC = os.getenv('MESSAGE_TOPIC', 'vulnerability_results')
+
+logging.basicConfig(level=logging.INFO)
+
+class QWriter(KafkaProducer):
+    def __init__(self, topic=ENGINE_RESULTS_TOPIC, bootstrap_servers=BOOTSTRAP_SERVERS, **kwargs):
+        super(QWriter, self).__init__(bootstrap_servers=bootstrap_servers, **kwargs)
+        self.results_topic = topic
+        logging.info("Created message producer:")
+        logging.info("Using BOOTSTRAP_SERVERS: %s", bootstrap_servers)
+        logging.info("Using ENGINE_RESULTS_TOPIC: %s", self.results_topic)
+
+    def send(self, msg):
+        data = json.dumps(msg).encode("utf8")
+        super(QWriter, self).send(self.results_topic, bytes(data))
+
+
+class QReader(KafkaConsumer):
+    def __init__(self, topic=ENGINE_RESULTS_TOPIC, group_id=GROUP_ID,
+                 bootstrap_servers=BOOTSTRAP_SERVERS, **kwargs):
+        super(QReader, self).__init__(topic,
+                                      bootstrap_servers=bootstrap_servers,
+                                      group_id=group_id,
+                                      auto_offset_reset='earliest',
```

Comment:
should be `'latest'` instead. Now in this code, every consumer starts processing of all the messages from the beginning of a queue.